update game docs

R3alCl0ud opened the pull request:
Added proper documentation for game field from the Presence Update Event.
the description for type might not be 100% accurate.

R3alCl0ud committed:
update game docs

Added proper documentation for game field from the Presence Update Event.

b1naryth1ef commented on docs/topics/Gateway.md:
```diff
@@ -563,10 +563,30 @@ A user's presence is their current state on a guild. This event is sent when a u
+| Field | Type | Description | Present |
+|-------|------|-------------|---------|
+| name | string | the game's name | Always |
+| type | ?integer | 0 is default, 1 is a twitch stream, 2 is a youtube stream  | Sometimes |
```
This should be broken out into a subtable as well "Game Types", also they key is as such:

| ID | Name | Desc |
|----|----------|---------|
| 0 | Game | "Playing {}" |
| 1 | Twitch | "Streaming {}" |

feel free to format in another way.

b1naryth1ef replied:
should also note we validate the URL for twitch

b1naryth1ef commented on docs/topics/Gateway.md:
```diff
@@ -563,10 +563,30 @@ A user's presence is their current state on a guild. This event is sent when a u
 | user | [user](#DOCS_USER/user-object) object | the user presence is being updated for |
 | roles | array of snowflakes | roles this user is in |
-| game | object | null, or an object containing one key of "name" |
+| game | ?[game](#DOCS_GATEWAY/game-object) object | the game being played/streamed, or null |
```
`an addition to the users status, generally showing their currently "activity"` or smth along those lines

R3alCl0ud committed:
made requested changes

moved type description to new subtable, Game Types. Also noted that url is validated when type is 1 (Twitch stream)

night commented on docs/topics/Gateway.md:
```diff
@@ -563,10 +563,38 @@ A user's presence is their current state on a guild. This event is sent when a u
+|----|------|-------------|
+| 0 | Game | "Playing {}" |
+| 1 | Twitch | "Streaming {}" |
+| 2 | YouTube | "Streaming {}" |
```
where did you get this?

`1` is Streaming.. not Twitch specifically.

DV8FromTheWorld replied:
Is `2` actually Youtube? 

If `1` is a generic "Streaming", Does it support anything other than Twitch urls currently?

b1naryth1ef replied:
No, we do consider it generic streaming but it currently only supports twitch. We can either name it twitch in these docs, which is IMO fine, or clarify a valid twitch url should be passed.

night commented on docs/topics/Gateway.md:
````diff
@@ -563,10 +563,38 @@ A user's presence is their current state on a guild. This event is sent when a u
+
+```json
+{
+	"name": "google chrome",
````
also, a better game name for streaming would be nice.. I don't think "google chrome" is a game.

R3alCl0ud committed:
changed example game name

MinnDevelopment commented on docs/topics/Gateway.md:
```diff
@@ -563,10 +563,38 @@ A user's presence is their current state on a guild. This event is sent when a u
+|-------|------|-------------|---------|
+| name | string | the game's name | Always |
+| type | ?integer | see [Game Types](#DOCS_GATEWAY/game-types)  | Sometimes |
+| url | ?string | stream url, is validated when type is 1 | When type is 1 or 2 |
```
Type 2 doesn't exist

R3alCl0ud committed:
updated Game Types

Removed type 2, leaving type 1 named  Twitch as Twitch is currently the only supported platform for streaming

MinnDevelopment commented on docs/topics/Gateway.md:
```diff
@@ -575,15 +575,14 @@ A user's presence is their current state on a guild. This event is sent when a u
 | ID | Name | Description |
 |----|------|-------------|
 | 0 | Game | "Playing Rocket League" |
 | 1 | Twitch | "Streaming Rocket League" |
```
As night already pointed out, this is `Stream` not `Twitch`

R3alCl0ud replied:
read commit description
> "Removed type 2, leaving type 1 named  Twitch as Twitch is currently the only supported platform for streaming"

R3alCl0ud replied:
also, 

> b1naryth1ef: "No, we do consider it generic streaming but it currently only supports twitch. We can either name it twitch in these docs, which is IMO fine, or clarify a valid twitch url should be passed."
